v1/s012-pyglet/pong.html

```diff
@@ -0,0 +1,616 @@
+<pre>
+import pyglet
+...
+window = pyglet.window.Window(width=SIRKA, height=VYSKA)
+pyglet.app.run()  # vse je nastaveno, at zacne hra
+</pre>
+                        Než začneme dělat interaktivní část hry reagující na vstupy
+                        od uživatele, je třeba umět vůbec vykreslit prvky na hrací
+                        ploše. Podobně jako jsme v přechozí lekci měli funkci
+                        <code>vykresli()</code>, která vykreslila hada, budeme mít
+                        v Pongu funkci stejného jména, která vykreslí prvky na hrací
+                        ploše.
+                    </div>
+                    <div> 
+                        Většina z tvarů jsou obdélníky, takže nejprve
+                        navrhněme funkci <code>nakresli_obdelnik</code>, která
+                        dostane čtveřici souřadnic a pomocí modulu <code>gl</code>
+                        z Pygletu vykreslí čtverec pomocí 2 trojúhelníků.
+<pre>
+from pyglet import gl
+...
+def nakresli_obdelnik(x1, y1, x2, y2):
+    """Nakresli obdelnik na dane souradnice
+
+    Nazorny diagram::
+
+         y2 - +-----+
+              |/////|
+         y1 - +-----+
+              :     :
+             x1    x2
+    """
+    # Tady pouzijeme volani OpenGL, ktere je pro nas zatim asi nejjednodussi
+    # na pouziti
+    gl.glBegin(gl.GL_TRIANGLE_FAN)   # zacni kreslit spojene trojuhelniky
+    gl.glVertex2f(int(x1), int(y1))  # vrchol A
+    gl.glVertex2f(int(x1), int(y2))  # vrchol B
+    gl.glVertex2f(int(x2), int(y2))  # vrchol C, nakresli trojuhelnik ABC
+    gl.glVertex2f(int(x2), int(y1))  # vrchol D, nakresli trojuhelnik BCD
+    # dalsi souradnice E by nakreslila trojuhelnik CDE, atd.
+    gl.glEnd()  # ukonci kresleni trojuhelniku
+</pre>
+                    </div>
+                    <div>
+                        Teď začneme pracovat na funkci <code>vykresli()</code>
+                        Nejprve ji vytvořte prázdnou a zaregistrujte ji
+                        v Pygletu na událost <code>on_draw</code>, jak jsme to 
+                        dělali na minulé lekci. To znamená, že se tato funkce
+                        zavolá pokaždé, když Pyglet překreslí okno. Pokud se
+                        mezitím např. změnila poloha míčku, funkce ho vykreslí
+                        o kousek jinde. Tím vlastně vytváříme dynamiku hry.
+                        Analogicky jsme to dělali s hadem, tady máme jen víc
+                        grafických prvků.
+<details class="solution notes"><h4>Řešení</h4>
+<pre>
+...
+def vykresli():
+    """Vykresli stav hry"""
+    gl.glClear(gl.GL_COLOR_BUFFER_BIT)  # smaz obsah okna (vybarvi na cerno)
+    gl.glColor3f(1, 1, 1)  # nastav barvu kresleni na bilou
+
+window = pyglet.window.Window(width=SIRKA, height=VYSKA)
+window.push_handlers(
+    on_draw=vykresli,  # na vykresleni okna pouzij funkci `vykresli`
+)
+pyglet.app.run()  # vse je nastaveno, at zacne hra
+</pre>
+</details>
```

Comment:
Já myslím že na tohle nejde přijít z informací řečených předtím, takže bych to nedával jako úkol k řešení.